docs/pages/guides/server-components.mdx

```diff
@@ -0,0 +1,400 @@
+---
+title: Using React Server Components in Expo Router apps
+sidebar_title: React Server Components
+description: Learn about rendering React components on the server in Expo.
+---
+
+import { Terminal } from '~/ui/components/Snippet';
+
+> Experimentally available in **SDK 52 and above**. This is a very early preview and not ready for production or wide use yet. This initial version is meant to help library authors understand universal React Server Components and add support in their libraries.
+
+Expo Router enables support for [React Server Components](https://react.dev/reference/rsc/server-components) on all platforms.
+
+React Server Components enable a number of exciting capabilities including:
+
+- Data fetching with async components and React Suspense.
+- Working with secrets and server-side APIs.
+- Server-side rendering (SSR) for SEO and performance.
+- Build-time rendering to remove unused JS code.
+
+This is an early preview of a feature that will be enabled by default in Expo Router.
+
+## Usage
+
+Your project must be using Expo Router and React Native new architecture. New Architecture is not supported in Expo Go so you'll need to create a development build or use the web.
+
+As of SDK 52, there's no strategy to migrate existing projects, we recommend starting from a new project.
+
+Enable the flag in the project Expo config:
+
+```json app.json
+{
+  "expo": {
+    "experiments": {
+      "reactServerComponents": true
+    }
+  }
+}
+```
+
+Ensure the entry module is `expo-router/entry` (default).
+
+Create an initial route `app/index.tsx`:
+
+```tsx app/index.tsx
+// This is a server component.
+import { Text } from 'react-native';
+
+export default function Index() {
+  return <Text>Hello, world!</Text>;
+}
+```
+
+## Moving between routes
+
+You can navigate between routes by using the `<Link />` API from Expo Router (not all props are supported yet).
+
+```tsx app/index.tsx
+import { Link } from 'expo-router';
+
+export default function Index() {
+  return (
+    <Link href="/second">
+      <Text>Hello, world!</Text>
+    </Link>
+  );
+}
+```
+
+## Layout Routes
+
+> Complex built-in layouts such as Stack, and Tabs are not supported yet. They'll fallback to a `<Slot />`.
+
+Layout routes (`app/_layout.js`) can be used to wrap all routes in a common layout.
```

Comment:
```suggestion
Layout routes (**app/_layout.tsx**) can wrap all routes in a common layout.
```

Since all examples use `tsx` as file extensions, let's stick to that here as well.